File: CodingSamples/Java/Inheritance/AbstractClassTest/src/SavingsAccount.java
package banking;

final class SavingsAccount extends Account {

	//field declared with final modifier cannot be reinitialized
	static final double MIN_BALANCE = 5000;

	SavingsAccount() {
		balance = MIN_BALANCE;
	}

	public void deposit(double amount) {
		balance += amount;
	}

	public void withdraw(double amount) throws InsufficientFundsException {
		if(balance - amount < MIN_BALANCE)
			throw new InsufficientFundsException();
		balance -= amount;
	}
}

File: CodingSamples/DotNet/Inheritance/InterfaceTest1/Banking/SavingsAccount.cs
namespace Banking;

sealed class SavingsAccount : Account, IProfitable
{
    const double MinBal = 5000;

    internal SavingsAccount()
    {
        Balance = MinBal;
    }

    public override void Deposit(double amount)
    {
        Balance += amount;
    }

    public override void Withdraw(double amount)
    {
        if(Balance - amount < MinBal)
            throw new InsufficientFundsException();
        Balance -= amount;
    }

    public double AddInterest(int months)
    {
        double interest = Balance * months * IProfitable.MinRate / 1200;
        Balance += interest;
        return interest;
    }
}